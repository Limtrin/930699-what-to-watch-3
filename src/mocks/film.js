export const Film = {
  id: `grand-budapest`,
  name: `The Grand Budapest Hotel`,
  genre: `Dramas`,
  year: 2019,
  image: `img/bohemian-rhapsody.jpg`,
  poster: `img/the-grand-budapest-hotel-poster.jpg`,
  cover: `img/bg-the-grand-budapest-hotel.jpg`,
  preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  time: `2h 22m`,
  rating: 9.5,
  votes: 123,
  director: `Frank Spaskukotski`,
  description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes).`,
  reviews: [
    {
      rating: 10,
      date: `March 15, 2019`,
      author: {
        id: 1,
        name: `Jane Doe`
      },
      text: `Fantastic!`
    },
    {
      rating: 10,
      date: `March 15, 2019`,
      author: {
        id: 1,
        name: `Jane Doe`
      },
      text: `Fantastic!`
    },
    {
      rating: 10,
      date: `March 15, 2019`,
      author: {
        id: 1,
        name: `Jane Doe`
      },
      text: `Fantastic!`
    }
  ],
  starring: [
    `Tim Robbins`,
    `Morgan Freeman`,
    `Bob Gunton`,
    `William Sadler`,
    `Clancy Brown`,
    `Gil Bellows`,
    `James Whitmore`
  ],
};
