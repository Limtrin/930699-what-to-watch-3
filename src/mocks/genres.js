
const FILMS_GENRES = [
  `All genres`,
  `Comedies`,
  `Crime`,
  `Documentary`,
  `Dramas`,
  `Horror`,
  `Kids & Family`,
  `Romance`,
  `Sci-Fi`,
  `Thrillers`,
];

export default FILMS_GENRES;
