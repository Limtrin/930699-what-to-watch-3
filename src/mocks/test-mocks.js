export const filmsList = [
  {
    id: `shutter-island`,
    name: `Shutter Island`,
    genre: `detective`,
    year: 2010,
    image: `img/shutter-island.jpg`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    cover: `img/bg-the-grand-budapest-hotel.jpg`,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    time: `2h 22m`,
    rating: 9.5,
    votes: 123,
    director: `Frank Spaskukotski`,
    description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes).`,
    reviews: [
      {
        rating: 10,
        date: `March 15, 2019`,
        author: {
          id: 1,
          name: `Jane Doe`
        },
        text: `Fantastic!`
      },
      {
        rating: 10,
        date: `March 15, 2019`,
        author: {
          id: 1,
          name: `Jane Doe`
        },
        text: `Fantastic!`
      },
      {
        rating: 10,
        date: `March 15, 2019`,
        author: {
          id: 1,
          name: `Jane Doe`
        },
        text: `Fantastic!`
      }
    ],
    starring: [
      `Tim Robbins`,
      `Morgan Freeman`,
      `Bob Gunton`,
      `William Sadler`,
      `Clancy Brown`,
      `Gil Bellows`,
      `James Whitmore`
    ],
  },
  {
    id: `snatch`,
    name: `Snatch`,
    genre: `comedy`,
    year: 2009,
    image: `img/snatch.jpg`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    cover: `img/bg-the-grand-budapest-hotel.jpg`,
    preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    time: `2h 22m`,
    rating: 9.5,
    votes: 123,
    director: `Frank Spaskukotski`,
    description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes).`,
    reviews: [
      {
        rating: 10,
        date: `March 15, 2019`,
        author: {
          id: 1,
          name: `Jane Doe`
        },
        text: `Fantastic!`
      },
      {
        rating: 10,
        date: `March 15, 2019`,
        author: {
          id: 1,
          name: `Jane Doe`
        },
        text: `Fantastic!`
      },
      {
        rating: 10,
        date: `March 15, 2019`,
        author: {
          id: 1,
          name: `Jane Doe`
        },
        text: `Fantastic!`
      }
    ],
    starring: [
      `Tim Robbins`,
      `Morgan Freeman`,
      `Bob Gunton`,
      `William Sadler`,
      `Clancy Brown`,
      `Gil Bellows`,
      `James Whitmore`
    ],
  }
];

export const film = {
  id: `war-of-the-worlds`,
  name: `War of the Worlds`,
  genre: `action`,
  year: 2008,
  image: `img/war-of-the-worlds.jpg`,
  poster: `img/the-grand-budapest-hotel-poster.jpg`,
  cover: `img/bg-the-grand-budapest-hotel.jpg`,
  preview: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  time: `2h 22m`,
  rating: 9.5,
  votes: 123,
  director: `Frank Spaskukotski`,
  description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes).`,
  reviews: [
    {
      rating: 10,
      date: `March 15, 2019`,
      author: {
        id: 1,
        name: `Jane Doe`
      },
      text: `Fantastic!`
    },
    {
      rating: 10,
      date: `March 15, 2019`,
      author: {
        id: 1,
        name: `Jane Doe`
      },
      text: `Fantastic!`
    },
    {
      rating: 10,
      date: `March 15, 2019`,
      author: {
        id: 1,
        name: `Jane Doe`
      },
      text: `Fantastic!`
    }
  ],
  starring: [
    `Tim Robbins`,
    `Morgan Freeman`,
    `Bob Gunton`,
    `William Sadler`,
    `Clancy Brown`,
    `Gil Bellows`,
    `James Whitmore`
  ],
};
