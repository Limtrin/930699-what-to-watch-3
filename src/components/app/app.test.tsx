import * as React from "react";
import * as renderer from "react-test-renderer";
import App from "./app";
import {filmsList, film} from "../../mocks/test-mocks";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import NameSpace from "../../reducer/name-space";
import {AuthorizationStatus} from "../../reducer/user/user";
import {BrowserRouter} from "react-router-dom";

const mockStore = configureStore([]);

it(`Render App`, () => {
  const store = mockStore({
    [NameSpace.USER]: {
      authorizationStatus: AuthorizationStatus.NO_AUTH,
      pending: true,
    },
    [NameSpace.DATA]: {
      genre: `All genres`,
      filmsList,
      filmsCurrent: filmsList,
      showedFilms: filmsList.slice(0, 8),
      filmsCount: 8,
      film,
    }
  });
  const tree = renderer
    .create(
        <Provider store={store}>
          <BrowserRouter>
            <App
              film={film}
            />
          </BrowserRouter>
        </Provider>
        , {
          createNodeMock: () => {
            return {};
          }
        }
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
