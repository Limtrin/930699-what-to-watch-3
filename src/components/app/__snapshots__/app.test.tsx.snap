// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render App 1`] = `
Array [
  <section
    className="movie-card"
  >
    <div
      className="movie-card__bg"
    >
      <img
        alt="War of the Worlds"
        src="img/bg-the-grand-budapest-hotel.jpg"
      />
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <header
      className="page-header movie-card__head"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link"
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="user-block"
      >
        <a
          href="/login"
          onClick={[Function]}
        >
          Sign In
        </a>
      </div>
    </header>
    <div
      className="movie-card__wrap"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster"
        >
          <img
            alt="The Grand Budapest Hotel poster"
            height="327"
            src="img/the-grand-budapest-hotel-poster.jpg"
            width="218"
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            War of the Worlds
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              action
            </span>
            <span
              className="movie-card__year"
            >
              2008
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <a
              className="btn btn--play movie-card__button"
              href="/films/war-of-the-worlds/player"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlinkHref="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </a>
            <a
              className="btn btn--list movie-card__button"
              href="/login"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="20"
                viewBox="0 0 19 20"
                width="19"
              >
                <use
                  xlinkHref="#add"
                />
              </svg>
              <span>
                My list
              </span>
            </a>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <div
    className="page-content"
  >
    <section
      className="catalog"
    >
      <h2
        className="catalog__title visually-hidden"
      >
        Catalog
      </h2>
      <ul
        className="catalog__genres-list"
      >
        <li
          className="catalog__genres-item catalog__genres-item--active"
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            All genres
          </a>
        </li>
        <li
          className="catalog__genres-item "
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            detective
          </a>
        </li>
        <li
          className="catalog__genres-item "
        >
          <a
            className="catalog__genres-link"
            href="#"
            onClick={[Function]}
          >
            comedy
          </a>
        </li>
      </ul>
      <div
        className="catalog__movies-list"
      >
        <article
          className="small-movie-card catalog__movies-card"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              height="175"
              onLoadStart={[Function]}
              onPlay={[Function]}
              poster="img/shutter-island.jpg"
              src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/shutter-island"
              onClick={[Function]}
            >
              Shutter Island
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              height="175"
              onLoadStart={[Function]}
              onPlay={[Function]}
              poster="img/snatch.jpg"
              src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="/films/snatch"
              onClick={[Function]}
            >
              Snatch
            </a>
          </h3>
        </article>
      </div>
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          Â© 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;
